File: src/二叉树/144.二叉树的前序遍历/144.二叉树的前序遍历.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []  # 数组存储遍历结果

        def tarversal(root):  # 构建递归函数,确定参数
            if root == None: return  # 确定终止条件， 确定返回值
            # 确当单层逻辑
            res.append(root.val)
            tarversal(root.left)
            tarversal(root.right)

        tarversal(root)
        return res

File: src/二叉树/144.二叉树的前序遍历/144.二叉树的前序遍历.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var preorderTraversal = function(root) {
    let res = new Array();  // 构造数组存放结果
    // 构造递归函数, 确定参数和返回值
    const traversal = function(root) {
        // 确定终止条件
        if(root === null){
            return
        };
        // 确定单层循环逻辑
        res.push(root.val)
        traversal(root.left)
        traversal(root.right)
    };
    traversal(root);
    return res
};

